pydantic/json_schema.py

```diff
@@ -461,6 +462,7 @@ def tagged_union_schema(self, schema: core_schema.TaggedUnionSchema) -> JsonSche
             if isinstance(v, (str, int)):
                 while isinstance(schema['choices'][v], (str, int)):
                     v = schema['choices'][v]
+                    assert isinstance(v, (int, str))
```

Comment:
this is probably more efficient to do as a cast on the previous line, not that it really matters

Reply:
asserts are nice in the sense that you can't lie about it, which is probably a good thing given that I'm kinda throwing these around willy nilly right now